eland/dataframe.py

```diff
@@ -569,6 +569,54 @@ def info_es(self):
 
         return buf.getvalue()
 
+    def es_query(self, query):
+        """Applies an Elasticsearch DSL query to the current DataFrame.
+
+        Parameters
+        ----------
+        query:
+            Dictionary of the Elasticsearch DSL query to apply
+
+        Returns
+        -------
+        eland.DataFrame:
+            eland DataFrame with the query applied
+
+        Examples
+        --------
+
+        Apply a `geo-distance query`_ to a dataset with a geo-point column ``geoip.location``.
+
+         .. _geo-distance query: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-geo-distance-query.html
+
+        >>> df = ed.DataFrame('localhost', 'ecommerce', columns=['customer_first_name', 'geoip.city_name'])
+        >>> df.es_query({"bool": {"filter": {"geo_distance": {"distance": "1km", "geoip.location": [25.3, 55.3]}}}})
```

Comment:
Other examples include results which pass `pytest --doctest-modules` - or skip test. Probably good to review doctests.

Reply:
Ah that's how you run the doc tests! Should I add that to the `test` nox task?

Reply:
Doc tests now run as a part of CI.